### ISSUE TITLE
[MJAR-228] Plugin generates wrong Class-Path in MANIFEST

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-jar-plugin#382]
**[Andres](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=ajaimes)** opened **[MJAR-228](https://issues.apache.org/jira/browse/MJAR-228?redirect=false)** and commented

Hello everyone,

Classpath is incorrectly generated using version 3.0.0. Take the following pom excerpt as an example:

```xml
<plugin>
    <groupId>org.apache.maven.plugins</groupId>
    <artifactId>maven-jar-plugin</artifactId>
    <version>2.4</version>
    <configuration>
        <archive>
            <manifest>
                <addClasspath>true</addClasspath>
                <classpathPrefix>lib/</classpathPrefix>
                <mainClass>com.example.Main</mainClass>
            </manifest>
        </archive>
    </configuration>
</plugin>
```

Version 2.4 generates the following correct MANIFEST:

```
Manifest-Version: 1.0
Archiver-Version: Plexus Archiver
Built-By: andres
Class-Path: lib/jersey-container-grizzly2-http-2.22.2.jar lib/javax.in
 ject-2.4.0-b34.jar lib/grizzly-http-server-2.3.23.jar lib/grizzly-htt
 p-2.3.23.jar lib/grizzly-framework-2.3.23.jar lib/jersey-common-2.22.
 2.jar lib/javax.annotation-api-1.2.jar lib/jersey-guava-2.22.2.jar li
 b/hk2-api-2.4.0-b34.jar lib/hk2-utils-2.4.0-b34.jar lib/aopalliance-r
 epackaged-2.4.0-b34.jar lib/hk2-locator-2.4.0-b34.jar lib/javassist-3
 .18.1-GA.jar lib/osgi-resource-locator-1.0.1.jar lib/jersey-server-2.
 22.2.jar lib/jersey-client-2.22.2.jar lib/jersey-media-jaxb-2.22.2.ja
 r lib/validation-api-1.1.0.Final.jar lib/javax.ws.rs-api-2.0.1.jar li
 b/jersey-media-moxy-2.22.2.jar lib/jersey-entity-filtering-2.22.2.jar
  lib/org.eclipse.persistence.moxy-2.6.0.jar lib/org.eclipse.persisten
 ce.core-2.6.0.jar lib/org.eclipse.persistence.asm-2.6.0.jar lib/javax
 .json-1.0.4.jar lib/solr-solrj-6.0.1.jar lib/commons-io-2.4.jar lib/h
 ttpclient-4.4.1.jar lib/httpcore-4.4.1.jar lib/httpmime-4.4.1.jar lib
 /zookeeper-3.4.6.jar lib/stax2-api-3.1.4.jar lib/woodstox-core-asl-4.
 4.1.jar lib/noggit-0.6.jar lib/jcl-over-slf4j-1.7.7.jar lib/slf4j-api
 -1.7.7.jar
Created-By: Apache Maven 3.3.9
Build-Jdk: 1.8.0_91
Main-Class: com.example.Main
```

Version 3.0.0 generates the following incorrect one:

```
Manifest-Version: 1.0
Archiver-Version: Plexus Archiver
Built-By: andres
Class-Path: lib/org/glassfish/jersey/containers/jersey-container-grizz
 ly2-http/2.22.2/jersey-container-grizzly2-http-2.22.2.jar lib/org/gla
 ssfish/hk2/external/javax.inject/2.4.0-b34/javax.inject-2.4.0-b34.jar
  lib/org/glassfish/grizzly/grizzly-http-server/2.3.23/grizzly-http-se
 rver-2.3.23.jar lib/org/glassfish/grizzly/grizzly-http/2.3.23/grizzly
 -http-2.3.23.jar lib/org/glassfish/grizzly/grizzly-framework/2.3.23/g
 rizzly-framework-2.3.23.jar lib/org/glassfish/jersey/core/jersey-comm
 on/2.22.2/jersey-common-2.22.2.jar lib/javax/annotation/javax.annotat
 ion-api/1.2/javax.annotation-api-1.2.jar lib/org/glassfish/jersey/bun
 dles/repackaged/jersey-guava/2.22.2/jersey-guava-2.22.2.jar lib/org/g
 lassfish/hk2/hk2-api/2.4.0-b34/hk2-api-2.4.0-b34.jar lib/org/glassfis
 h/hk2/hk2-utils/2.4.0-b34/hk2-utils-2.4.0-b34.jar lib/org/glassfish/h
 k2/external/aopalliance-repackaged/2.4.0-b34/aopalliance-repackaged-2
 .4.0-b34.jar lib/org/glassfish/hk2/hk2-locator/2.4.0-b34/hk2-locator-
 2.4.0-b34.jar lib/org/javassist/javassist/3.18.1-GA/javassist-3.18.1-
 GA.jar lib/org/glassfish/hk2/osgi-resource-locator/1.0.1/osgi-resourc
 e-locator-1.0.1.jar lib/org/glassfish/jersey/core/jersey-server/2.22.
 2/jersey-server-2.22.2.jar lib/org/glassfish/jersey/core/jersey-clien
 t/2.22.2/jersey-client-2.22.2.jar lib/org/glassfish/jersey/media/jers
 ey-media-jaxb/2.22.2/jersey-media-jaxb-2.22.2.jar lib/javax/validatio
 n/validation-api/1.1.0.Final/validation-api-1.1.0.Final.jar lib/javax
 /ws/rs/javax.ws.rs-api/2.0.1/javax.ws.rs-api-2.0.1.jar lib/org/glassf
 ish/jersey/media/jersey-media-moxy/2.22.2/jersey-media-moxy-2.22.2.ja
 r lib/org/glassfish/jersey/ext/jersey-entity-filtering/2.22.2/jersey-
 entity-filtering-2.22.2.jar lib/org/eclipse/persistence/org.eclipse.p
 ersistence.moxy/2.6.0/org.eclipse.persistence.moxy-2.6.0.jar lib/org/
 eclipse/persistence/org.eclipse.persistence.core/2.6.0/org.eclipse.pe
 rsistence.core-2.6.0.jar lib/org/eclipse/persistence/org.eclipse.pers
 istence.asm/2.6.0/org.eclipse.persistence.asm-2.6.0.jar lib/org/glass
 fish/javax.json/1.0.4/javax.json-1.0.4.jar lib/org/apache/solr/solr-s
 olrj/6.0.1/solr-solrj-6.0.1.jar lib/commons-io/commons-io/2.4/commons
 -io-2.4.jar lib/org/apache/httpcomponents/httpclient/4.4.1/httpclient
 -4.4.1.jar lib/org/apache/httpcomponents/httpcore/4.4.1/httpcore-4.4.
 1.jar lib/org/apache/httpcomponents/httpmime/4.4.1/httpmime-4.4.1.jar
  lib/org/apache/zookeeper/zookeeper/3.4.6/zookeeper-3.4.6.jar lib/org
 /codehaus/woodstox/stax2-api/3.1.4/stax2-api-3.1.4.jar lib/org/codeha
 us/woodstox/woodstox-core-asl/4.4.1/woodstox-core-asl-4.4.1.jar lib/o
 rg/noggit/noggit/0.6/noggit-0.6.jar lib/org/slf4j/jcl-over-slf4j/1.7.
 7/jcl-over-slf4j-1.7.7.jar lib/org/slf4j/slf4j-api/1.7.7/slf4j-api-1.
 7.7.jar
Created-By: Apache Maven 3.3.9
Build-Jdk: 1.8.0_91
Main-Class: com.example.Main
```

Thanks

---

**Affects:** 3.0.0

**Issue Links:**
- [MJAR-229](https://issues.apache.org/jira/browse/MJAR-229) Upgrade maven-archiver to 3.1.1
 (_**"is blocked by"**_)

**Remote Links:**
- [GitHub Pull Request #39
](https://github.com/apache/maven-jar-plugin/pull/39)

**Comment by jira-importer** [comment]
**[Jonathan Haber](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=jhaber)** commented

I believe this is a duplicate of MJAR-223

**Comment by jira-importer** [comment]
**[Karl Heinz Marbaise](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=khmarbaise)** commented

Fixed in [r1749027](http://svn.apache.org/r1749027)
